expo/scheduler/views.py

```diff
@@ -593,117 +598,143 @@ def contact_info(request,
-                  'Event']
-        from gbe_forms_text import volunteer_interests_options
-        volunteer_categories = dict(volunteer_interests_options)
-        contact_info = []
-        for c in contacts:
-            profile = c.item.profile
-            event = c.allocations.first().event
-            try:
-                parent_event = event.container_event.parent_event
-            except:
-                parent_event = event
+def contact_performers(conference=None):
+    if not conference:
+        conference = get_current_conference()
+    from gbe.models import Act
+    contacts = [act.actitem_ptr for act in Act.objects.filter(
+        conference=conference)]
+    header = ['Act',
+              'Performer',
+              'Profile',
+              'email',
+              'Phone',
+              'status',
+              'Show(s)',
+              'Rehearsal(s)']
+    contact_info = []
+    for c in contacts:
+        act = c.as_subtype
+        performer = act.performer
+        contact_info.append(
+            [act.title,
+             str(performer),
+             str(performer.contact),
+             act.contact_email,
+             act.performer.contact.phone,
+             act.accepted,
+             ",".join(map(str, act.get_scheduled_shows())),
+             ",".join(map(str, act.get_scheduled_rehearsals()))]
+        )
+    return header, contact_info
+
+
+def contact_volunteers(conference=None):
+    header = ['Name',
+              'Phone',
+              'Email',
+              'Volunteer Category',
+              'Volunteer Role',
+              'Event']
+    volunteer_categories = dict(volunteer_interests_options)
+    if not conference:
+        conference = get_current_conference()
+    from gbe.models import Volunteer
+    contacts = filter(lambda worker: worker.allocations.count() > 0,
+                      [vol.profile.workeritem_ptr.worker_set.first() for vol in
+                       Volunteer.objects.filter(conference=conference)
+                       if vol.profile.workeritem_ptr.worker_set.exists()])
+
+    volunteers = Volunteer.objects.filter(conference=conference).annotate(
+        Count('profile__workeritem_ptr__worker')).order_by(
+            '-profile__workeritem_ptr__worker__count')
+
+    contact_info = []
+    for v in volunteers:
+        profile = v.profile
+        for worker in profile.workeritem_ptr.worker_set.all():
+            for allocation in worker.allocations.all():
+                try:
+                    container = allocation.event.container_event
+                    parent_event = container.parent_event
+                except:
+                    parent_event = allocation.event
+                contact_info.append(
+                    [profile.display_name,
+                     profile.phone,
+                     profile.contact_email,
+                     volunteer_categories.get(
+                         allocation.event.as_subtype.volunteer_category, ''),
```

Comment:
Something isn't working here... I'm seeing all sorts of 2015 events.  For example - Betty is booked for REGISTRATION (not Registration) - which was 2015, and there's multiple members of tech crews working on the '15 shows.  I'd like to know this one is eliminated before approving, since it could cause confusion with the more subtly named events (like registration and security)